qemu/tests/block_resize.py

```diff
@@ -1,171 +1,148 @@
 
-    1) Start guest with data image and check the data image size.
-    2) Enlarge(or Decrease) the data image and check it in guest.
+    1) Start guest with data disk or system disk.
+    2) Do format disk in guest if needed.
+    3) Enlarge the data disk image from qemu monitor.
+    4) Extend data disk partition/file-system in guest.
+    5) Verify the data disk size match expected size.
+    6) Reboot the guest.
+    7) Do iozone test.
+    8) Shrink data disk partition/file-system in guest.
+    9) Shrink data disk image from qemu monitor.
+    10) Verify the data disk size match expected size.
+    11) Reboot the guest.
+    12) Do iozone test.
 
     :param test: QEMU test object
     :param params: Dictionary with the test parameters
     :param env: Dictionary with test environment.
     """
-    def get_block_size(session, block_cmd, block_pattern):
-        """
-        Get block size inside guest.
-        """
-        output = session.cmd_output(block_cmd)
-        block_size = re.findall(block_pattern, output)
-        if block_size:
-            if not re.search("[a-zA-Z]", block_size[0]):
-                return int(block_size[0])
-            else:
-                return float(utils_misc.normalize_data_size(block_size[0],
-                                                            order_magnitude="B"))
-        else:
-            test.error("Can not find the block size for the"
-                       " deivce. The output of command"
-                       " is: %s" % output)
-
-    def compare_block_size(session, block_cmd, block_pattern):
+    def verify_disk_size(session, os_type, disk):
         """
-        Compare the current block size with the expected size.
         """
         global current_size
-        current_size = get_block_size(session,
-                                      block_size_cmd, block_size_pattern)
+        current_size = utils_disk.get_disk_size(session, os_type, disk)
+        accept_ratio = float(params.get("accept_ratio", 0))
         if (current_size <= block_size and
                 current_size >= block_size * (1 - accept_ratio)):
             logging.info("Block Resizing Finished !!! \n"
                          "Current size %s is same as the expected %s",
                          current_size, block_size)
             return True
-        return
 
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
     timeout = float(params.get("login_timeout", 240))
+    disk = 0
     driver_name = params.get("driver_name")
-    session = vm.wait_for_login(timeout=timeout)
-
-    if params.get("os_type") == "windows" and driver_name:
-        session = utils_test.qemu.windrv_check_running_verifier(session, vm,
-                                                                test, driver_name,
-                                                                timeout)
+    os_type = params["os_type"]
+    fstype = params.get("fstype")
+    labeltype = params.get("labeltype", "msdos")
```

Comment:
As we discussed before, not support gpt for linux. Suggest to set it in cfg.
e.g.
Linux:
    labeltype = msdos

Reply:
that's ok, will update~

Reply:
updated, thanks~